wrapper/src/main/java/software/amazon/jdbc/plugin/staledns/AuroraStaleDnsHelper.java

```diff
@@ -48,16 +48,29 @@ public class AuroraStaleDnsHelper {
   private InetAddress writerHostAddress = null;
   private static final int RETRIES = 3;
 
+  private static final String POSTGRESQL_READONLY_QUERY = "SELECT rt.server_id, CASE\n"
+      + "           WHEN 'MASTER_SESSION_ID' = rt.session_id THEN 'false'\n"
+      + "           ELSE 'true'\n"
+      + "           END AS is_reader\n"
+      + "FROM aurora_replica_status() rt,\n"
+      + "     aurora_db_instance_identifier() di\n"
+      + "WHERE rt.server_id = di;";
+
```

Comment:
`select pg_is_in_recovery () as is_reader;` is much simpler

Reply:
Thank you